openmicroscopy/openmicroscopy
# Django18 upgrade

This PR helps choosing right Django version depends on Python.
 - Python 2.6 Django 1.6
 - Python 2.7 Django 1.8

The aim of this PR is:
 - to deploy Omero.web using both python versions,
 - clarify use of ``request.GET`` and ``request.POST``
 - fix deprecated warnings.


To test:
- check if https://ci.openmicroscopy.org/view/DEV/job/OMERO-DEV-merge-daily/
  - OMERO-DEV-merge-integration-Python27 -> https://cowfish/integration
  - OMERO-DEV-merge-integration - > https://trout/integration
  are green
- log in to both Omero.web

To test web plugin:
- uninstall django
- bin/omero web start should tells ``Django is not installed!``.

-----------

**Known issues:**
 - request.REQUEST still in webgateway_cache. That will be fixed in another PR if ok

---------------

**To test ui:**

**testing servers: Django 1.6 on eel/merge and Django 1.8 on cowfish/integration
when reporting bugs please indicate which server was used and exact timestamp when error occurred**

- feedback: send comment and error
 - go to https://eel and cowfish/merge or integration/feedback/comment/ and send comment.
 - go to https://eel and cowfish/merge or integration/weberror/error_500/ and submit error.
- login:
 - go to https://eel and cowfish/merge or integration/webclient/login/?url=%2Fmerge%2Fwebclient%2Fuserdata%2F%3Fexperimenter%3D-1, log in, you should be redirected to https://eel and cowfish/merge or integration/webclient/userdata/?experimenter=-1
- webadmin:
 - Reset Password https://www.openmicroscopy.org/private/ome-internal/testing_scenarios/Resetpwd.html
 - add new user / group https://www.openmicroscopy.org/private/ome-internal/testing_scenarios/AdminGroupUserCreate.html (handled by robot test)
 - add new user / group https://www.openmicroscopy.org/private/ome-internal/testing_scenarios/AdminGroupUserEdit.html (handled by robot test)
 - edit my account (handled by robot test)
 - crop avatar
- webclient:
 - check direct link takes to the image webclient/userdata/?show=image-123
 - search using global searching as well as advance searching
 - annotation (test with multiple selected images) https://www.openmicroscopy.org/private/ome-internal/testing_scenarios/Annotate.html
 - create Project, Dataset, Screen
 - create and edit Share https://www.openmicroscopy.org/private/ome-internal/testing_scenarios/Share.html
 - edit channel name
 - rendering settings: copy, paste, save, apply to all 
 - download:
   - attachement
    - original file
    - export to
- scripts https://www.openmicroscopy.org/private/ome-internal/testing_scenarios/ScriptsClient.html
